### PR TITLE
Clean the application name from none-alphanumeric characters for the publisher apps

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

When publishing legacy `/v1/events`, the Application name must be cleaned from none-alphanumeric characters for:

- The `event-publisher-nats` app.
- The `event-publisher-proxy` app

**Related issue(s)**
#10218

### CODE REVIEW

File: components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go [hunk 1]
```diff
@@ -0,0 +1,329 @@
+package handlertest
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go:1]: How about using `handler_test.go` as the name of the file?

File: components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+	}
+}
+
+func NewApplicationListerOrDie(ctx context.Context, appName string) *application.Lister {
+	app := applicationtest.NewApplication(appName, nil)
+	appLister := fake.NewListerOrDie(ctx, app)
+	return appLister
+}
+
+// common test-cases for the proxy-handler and nats-handler
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go:117]: Let's use one between, proxy-handler or http-handler?

File: components/event-publisher-proxy/pkg/handler/nats/handler.go [hunk 3]
```diff
@@ -43,8 +43,8 @@ type NatsHandler struct {
 }
 
 // NewHandler returns a new Handler instance for the Event Publisher Proxy.
-func NewNatsHandler(receiver *receiver.HttpMessageReceiver, sender *sender.NatsMessageSender, requestTimeout time.Duration, legacyTransformer *legacy.Transformer, opts *options.Options, subscribedProcessor *subscribed.Processor, logger *logrus.Logger) *NatsHandler {
-	return &NatsHandler{
+func NewNatsHandler(receiver *receiver.HttpMessageReceiver, sender *sender.NatsMessageSender, requestTimeout time.Duration, legacyTransformer *legacy.Transformer, opts *options.Options, subscribedProcessor *subscribed.Processor, logger *logrus.Logger) *Handler {
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/handler/nats/handler.go:46]: `nats.NewHandler` is also expressive. Nats in `NewNatsHandler` seems redundant.

File: components/event-publisher-proxy/pkg/legacy-events/legacy.go [hunk 4]
```diff
@@ -72,20 +76,17 @@ func (t Transformer) checkParameters(parameters *apiv1.PublishEventParametersV1)
...
 		return nil
 	}
-	appName := ParseApplicationNameFromPath(request.URL.Path)
-	var err error
+
 	parameters := &apiv1.PublishEventParametersV1{}
 	decoder := json.NewDecoder(request.Body)
-	err = decoder.Decode(&parameters.PublishrequestV1)
-	if err != nil {
+	if err := decoder.Decode(&parameters.PublishrequestV1); err != nil {
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/legacy-events/legacy.go:89]: Is it possible to change the variable `PublishrequestV1` to `PublishRequestV1` in this PR?

File: components/event-publisher-proxy/cmd/event-publisher-proxy/main.go [hunk 5]
```diff
@@ -1,21 +1,24 @@
...
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
 	"github.com/kelseyhightower/envconfig"
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/application"
 	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/env"
-	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/handler"
+	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/handler/http"
```
Comment (sayanh) [inline components/event-publisher-proxy/cmd/event-publisher-proxy/main.go:13]: Here, http handler is basically the BEB handler. We also have [proxy-handler](https://github.com/kyma-project/kyma/pull/10523/files?file-filters%5B%5D=.go&file-filters%5B%5D=.mod&file-filters%5B%5D=.sum&hide-deleted-files=true#diff-b0e8438d638396e1f6e1719eee9dd43c37a3184409727b931116557917db828dR117) referring to this handler. Let's be consistent.

File: components/event-publisher-proxy/pkg/handler/nats/handler.go [hunk 6]
```diff
@@ -99,8 +99,8 @@ func (h *NatsHandler) ServeHTTP(writer http.ResponseWriter, request *http.Reques
 	return
 }
 
-func (h *NatsHandler) publishLegacyEventsAsCE(writer http.ResponseWriter, request *http.Request) {
-	event := h.LegacyTransformer.TransformsLegacyRequestsToCE(writer, request)
+func (h *Handler) publishLegacyEventsAsCE(writer http.ResponseWriter, request *http.Request) {
+	event := h.LegacyTransformer.TransformLegacyRequestsToCE(writer, request)
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/handler/nats/handler.go:103]: (nit)`Transformer transforms` is grammatically correct. What was your rationale to change it to `transform`?

File: components/event-publisher-proxy/pkg/handler/nats/handler_test.go [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+		messageReceiver := receiver.NewHttpMessageReceiver(test.natsConfig.Port)
+		assert.NotNil(t, messageReceiver)
+
+		// create a Nats sender
+		natsUrl := test.natsServer.ClientURL()
+		assert.NotEmpty(t, natsUrl)
+		msgSender := sender.NewNatsMessageSender(ctx, natsUrl, test.logger)
+
+		// configure legacyTransformer
+		appLister := handlertest.NewApplicationListerOrDie(ctx, applicationName)
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/handler/nats/handler_test.go:91]: Can we scoop informer set up and make it common helpers? It is applicable for HTTP handler as well. `exec` is huge IMO.

File: components/event-publisher-proxy/pkg/handler/nats/handler_test.go [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+		legacyTransformer := legacy.NewTransformer(
+			test.natsConfig.ToConfig().BEBNamespace,
+			test.natsConfig.ToConfig().EventTypePrefix,
+			appLister,
+		)
+
+		// Setting up fake informers
+		scheme := runtime.NewScheme()
```
Comment (sayanh) [inline components/event-publisher-proxy/pkg/handler/nats/handler_test.go:214]: Informer set up is duplicated from the testing of `/publish`.

File: components/event-publisher-proxy/pkg/application/lister.go [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+	applicationv1alpha1 "github.com/kyma-project/kyma/components/application-operator/pkg/apis/applicationconnector/v1alpha1"
+	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/informers"
+)
+
+type Lister struct {
+	lister cache.GenericLister
+}
+
+func NewLister(ctx context.Context, client dynamic.Interface) *Lister {
+	const defaultResync = 10 * time.Second
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/application/lister.go:25]: This is not used outside the package. No need to make it global.

File: components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go (hunk 1 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go:1]: This file contains utils for handler testing. Same pattern is used in golang itself, check here: https://github.com/golang/go/blob/master/src/net/http/httptest/httptest.go.

File: components/event-publisher-proxy/pkg/handler/nats/handler.go (hunk 6 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/nats/handler.go:103]: Why introducing a new naming convention? Let's be consistent and use the same naming convention everywhere.

File: components/event-publisher-proxy/pkg/legacy-events/legacy.go (hunk 4 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/legacy-events/legacy.go:89]: I agree this should change, but please let's not extend the scope of this ticket.

File: components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go (hunk 2 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go]: Done

File: components/event-publisher-proxy/pkg/handler/nats/handler.go (hunk 3 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/nats/handler.go]: Done

File: components/event-publisher-proxy/pkg/handler/nats/handler_test.go (hunk 7 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/nats/handler_test.go]: Done.

File: components/event-publisher-proxy/pkg/handler/nats/handler_test.go (hunk 8 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/nats/handler_test.go]: Done

File: components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go (hunk 1 above)
Comment (sayanh) [reply components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go:1]: A comment of this [sort](https://github.com/golang/go/blob/master/src/net/http/httptest/httptest.go#L5)would make sense then.

File: components/event-publisher-proxy/pkg/handler/nats/handler.go (hunk 6 above)
Comment (sayanh) [reply components/event-publisher-proxy/pkg/handler/nats/handler.go:103]: `TransformLegacyRequestsToCE` is newly introduced. Why `TransformsLegacyRequestsToCE` was not ok?

**Review by sayanh — DISMISSED** [review]

File: components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go (hunk 1 above)
Comment (marcobebway) [reply components/event-publisher-proxy/pkg/handler/handlertest/handlertest.go:2]: Done

**Review by sayanh — APPROVED** [review]

**Pull request merged**